sensor/kubernetes/localscanner/service_certificates_repository_test.go

```diff
@@ -213,13 +213,13 @@ func (s *serviceCertificatesRepoSecretsImplSuite) TestGetSecretDataMissingKeysSu
 	}
 }
 
-func (s *serviceCertificatesRepoSecretsImplSuite) TestEnsureCertsUnknownServiceTypeFailure() {
+func (s *serviceCertificatesRepoSecretsImplSuite) TestEnsureCertsUnknownServiceTypeIgnores() {
 	fixture := s.newFixture(certSecretsRepoFixtureConfig{})
-	s.getFirstServiceCertificate(fixture.certificates).ServiceType = anotherServiceType
+	s.getFirstServiceCertificate(fixture.certificates).ServiceType = unknownServiceType
 
 	err := fixture.repo.ensureServiceCertificates(context.Background(), fixture.certificates)
 
-	s.Error(err)
+	s.NoError(err)
```

Comment:
I would add a comment here explaining why we don't consider this to be an error and what is happening instead (logs).

Reply:
`ensureServiceCertificates` is no longer returns an error if there is unknown service type for certificate. Instead it logs a warning. The method is not atomic so even though there is no unknown service type error anymore the method behaviour did not change. So if you pass 5 service types and one of them is unknown then method persists 4 and ignores unknown one. It used to return an error in such case and this PR changes error to warning semantic only

Reply:
Added additional secret check in the test to check that unknown type is really ignored